### PR TITLE
Add release steps to Github CI

### PR DESCRIPTION
**Pull request by tvinagre**

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+jobs:
+  create-release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: taiki-e/create-gh-release-action@v1
+        with:
+          draft: true
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (kayibal) [inline .github/workflows/release.yaml:23]: So this creates a release if we push a tag right? And what if the release is already present, e.g. if I create a release via the UI?

File: Dockerfile [hunk 2]
```diff
@@ -29,6 +29,7 @@ RUN rustup target add wasm32-unknown-unknown && ./stable-build.sh
 FROM debian:bookworm
 WORKDIR /opt/tycho-indexer
 COPY --from=build /build/target/release/tycho-indexer ./tycho-indexer
+COPY --from=build /build/target/release/tycho-client ./tycho-client
```
Comment (kayibal) [inline Dockerfile:32]: Do we still need this then?

File: stable-build.sh [hunk 3]
```diff
@@ -19,3 +20,6 @@ done
 
 echo "Building tycho: $TYCHO_INDEXER"
 cargo build --package $TYCHO_INDEXER --release
+
+echo "Building tycho-client: $TYCHO_CLIENT}"
+cargo build --package $TYCHO_CLIENT --release
```
Comment (kayibal) [inline stable-build.sh:25]: And this?

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# When a release is created, this workflow will build and upload the tycho-client binary to the release.
+name: Release
+
+on:
+  release:
+    types: [ created ]
+  push:
+    tags:
+      - v[0-9]+.*
```
Comment (kayibal) [inline .github/workflows/release.yaml]: I our tags don't have the `v` prefix

Comment (louise-poole) [reply .github/workflows/release.yaml]: Yeah - I think only frontend has this...

File: Dockerfile (hunk 2 above)
Comment (tvinagre) [reply Dockerfile:32]: Ah yeah I'd keep it - you can build and copy the binaries to linux-arm this way locally very easy

File: stable-build.sh (hunk 3 above)
Comment (tvinagre) [reply stable-build.sh:25]: Same for linux-arm

**Pull request merged**